backend/audit/test_schemas.py

```diff
@@ -403,77 +391,52 @@ def test_all_booleans(self):
             "is_aicpa_audit_guide_included",
             "is_low_risk_auditee",
         ]
-
-        for field in boolean_fields:
-            for value in [True, False]:
+        for value in [True, False]:
+            for field in boolean_fields:
                 simple_case[field] = value
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
-        boolean_fields.append("is_sp_framework_required")
-        simple_case = jsoncopy(self.SIMPLE_CASE[1])
-
-        for field in boolean_fields:
-            for value in [True, False]:
-                simple_case[field] = value
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
+                validate(simple_case, schema)
 
     def test_all_gaap_results(self):
-        simple_case = jsoncopy(self.SIMPLE_CASE[0])
-
-        for _ in range(5):
-            for n in range(1, 5):
-                ls = random.sample(self.ALL_OPINIONS, n)
-                simple_case["gaap_results"] = ls
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
-
-    def test_all_framework_basis(self):
-        simple_case = jsoncopy(self.SIMPLE_CASE[1])
-
-        sp_framework_basis = [
-            "cash_basis",
-            "tax_basis",
-            "contractual_basis",
-            "other_basis",
+        schema = self.AUDIT_INFO_SCHEMA
+        simple_case = jsoncopy(self.SIMPLE_CASE)
+        gaap_results = [
+            "unmodified_opinion",
+            "qualified_opinion",
+            "adverse_opinion",
+            "disclaimer_of_opinion",
+            "not_gaap",
         ]
 
-        for _ in range(5):
-            for n in range(1, 5):
-                opinions = random.sample(self.ALL_OPINIONS, n)
-                basis = random.sample(sp_framework_basis, n)
-                simple_case["sp_framework_opinions"] = opinions
-                simple_case["sp_framework_basis"] = basis
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
+        for result in gaap_results:
+            simple_case["gaap_results"] = [result]
+            validate(simple_case, schema)
```

Comment:
This doesn't work because it's changing the `gaap_results` value for a case that explicitly doesn't include the required property.